### PR TITLE
ARROW-13006: [C++][Gandiva] Implement BASE64 and UNBASE64 Hive functions on Gandiva

### PR DESCRIPTION
**Pull request by jpedroantunes**
Implement BASE64 and UNBASE64 Hive functions on Gandiva

### CODE REVIEW

**Comment by jpedroantunes** [comment]
@jvictorhuguenin can you review it, please?

**Review by jvictorhuguenin — CHANGES_REQUESTED** [review]
Add some projector tests for the added functions, please.

File: cpp/src/gandiva/gdv_function_stubs.cc [hunk 1]
```diff
@@ -302,6 +303,61 @@ char* gdv_fn_dec_to_string(int64_t context, int64_t x_high, uint64_t x_low,
...
+    gdv_fn_context_set_error_msg(context, "Buffer length can not be negative");
+    *out_len = 0;
+    return "";
+  }
+  if (in_len == 0) {
+    *out_len = 0;
+    return "";
+  }
+  // use arrow method to decode base64 string
+  std::string decoded_str = arrow::util::base64_decode(in);
```
Comment (projjal) [inline cpp/src/gandiva/gdv_function_stubs.cc:347]: This seems incorrect. It will assume null terminated string

Comment (jpedroantunes) [reply cpp/src/gandiva/gdv_function_stubs.cc]: You are right! I built the "in" string now considering its length on it. Now it not assumes the nulltermination

**Review by jvictorhuguenin — APPROVED** [review]

**Review by projjal — APPROVED** [review]

**Pull request closed without merging**